zanata-model/src/main/java/org/zanata/model/type/RequestStateTypeDescriptor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.model.type;
+
+/**
+ * Convert between a single character and a RequestState, used by
+ * {@link RequestStateType}.
+ */
+public class RequestStateTypeDescriptor
+        extends SingleCharEnumTypeDescriptor<RequestState> {
+
+    public static final RequestStateTypeDescriptor INSTANCE =
+            new RequestStateTypeDescriptor();
+
+    protected RequestStateTypeDescriptor() {
+        super(RequestState.class);
+    }
+
+    @Override
+    char getIdentifyingChar(RequestState value) {
+        return value.getInitial();
+    }
+
+    @Override
+    RequestState valueOf(String string) {
+        if (string.length() == 1) {
+            return RequestState.valueOf(string.charAt(0));
```

Comment:
Right, sorry. But some of the knowledge about this conversion does seem to be quite spread out. 

There should also only be one place which converts a String to RequestState. We have several places which call `charAt(0)` and then call `valueOf(char)`. I suggest adding a `valueOf(String)` method to RequestState, and using it everywhere instead of `RequestState.valueOf(string.charAt(0))`.